backend/apps/notificationpref/services.py

```diff
@@ -0,0 +1,50 @@
+from apps.notificationpref.models import UserNotificationPreference
+from apps.users.models import BaseUser
+from apps.communities.models import Community
+
+def notification_preference_create(
+    *, user: BaseUser, subscribed_communities: list = [], event_updates: bool = True,
+    post_notifications: bool = True, announcements: bool = True, email_notifications: bool = True,
+    in_app_notifications: bool = True
+) -> UserNotificationPreference:
+    
+    preference, created = UserNotificationPreference.objects.get_or_create(
```

Comment:
I think it should just be using .create() and then leave updating functionality to the _update service